### PR TITLE
Add the new dry strategy

### PR DESCRIPTION
**Pull request by roeiK-wix**
Hey, we finished refactoring the dry flag according to what we agreed on.  
Flag options:  
--dry=actions - Not executing any actions in the method and rollback sections.  
--dry=probes - Not executing any probes in the method and rollback sections.  
--dry=activities - Not executing any actions and probes in the method and rollback sections. (the same as the old dry flag)  
--dry=pause - Not executing any pause.  
During the implementation, we found that the old and now the new dry flag are just working inside the method section.  
But in all the other sections such as the steady-state and in the controls, the activities are still working as normal (meaning no dry)  
We wanted to know if this is the logic you meant to when you thought about adding a dry flag?  
Also, we found that the probes aren't returning any value, (as expected to a function that doesn't been called) so it always returns None and so it's always false tolerance.
In this new implementation, we kept the old logic and added the new logic with actionless, probeless, and pauseless.
Also, review this PR -> chaostoolkit/chaostoolkit#220
We didn't add any tests for this new logic because we wanted you to first review this PR and then we will add the tests.  
Thanks.

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -635,3 +712,6 @@ def test_remove_settings_entry():
             ],
         )
         assert result.exit_code == 1
+if __name__ == "__main__":
+    log_file = tempfile.NamedTemporaryFile()
+    test_dry_activities(log_file)
```
Comment (CharlieMoon37) [inline tests/test_cli.py:717]: The `log_file` is already created for you in [`tests/conftest.py`](https://github.com/chaostoolkit/chaostoolkit/blob/master/tests/conftest.py) so you don't need to recreate it

```suggestion

```

File: tests/test_cli.py [hunk 2]
```diff
@@ -18,6 +18,8 @@
 )
 from chaoslib.settings import CHAOSTOOLKIT_CONFIG_PATH
 from click.testing import CliRunner
+import tempfile
+
```
Comment (CharlieMoon37) [inline tests/test_cli.py:22]: The `log_file` is already created for you in [`tests/conftest.py`](https://github.com/chaostoolkit/chaostoolkit/blob/master/tests/conftest.py) so you don't need to recreate it

```suggestion
```

**Review by CharlieMoon37 — CHANGES_REQUESTED** [review]
Just a couple of things ...

File: tests/test_cli.py (hunk 1 above)
Comment (ciaranevans) [reply tests/test_cli.py:717]: This whole block isn't needed also as `pytest` does not use `if __name__ == "__main__"` to run tests

Comment (roeiK-wix) [reply tests/test_cli.py:717]: Yep, it was just for my tests. missed that before i pushed.

**Review by CharlieMoon37 — DISMISSED** [review]

**Pull request merged**